server/src/main/java/org/opensearch/index/translog/transfer/TranslogCheckpointSnapshotTransferManager.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.translog.transfer;
+
+import org.opensearch.action.LatchedActionListener;
+import org.opensearch.common.blobstore.BlobPath;
+import org.opensearch.common.blobstore.stream.write.WritePriority;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Translog transfer manager to transfer {@link TranslogCheckpointSnapshot}
+ *
+ * @opensearch.internal
+ */
+public interface TranslogCheckpointSnapshotTransferManager {
+
+    public void transferTranslogCheckpointSnapshot(
+        Set<TranslogCheckpointSnapshot> toUpload,
+        Map<Long, BlobPath> blobPathMap,
+        LatchedActionListener<TranslogCheckpointSnapshot> latchedActionListener,
+        WritePriority writePriority
+    ) throws Exception;
```

Comment:
Why are we creating another class to define the behaviour of already existing class `TranslogTransferManager`. 

We should have the existing `TranslogTransferManager` as `BaseTranslogTransferManager`, and have have 2 implementations of `BaseTranslogTransferManager`. The `TranslogTransferManagerFactory` should decide the implementation. Please have concise names for the 2 implementations.

Reply:
Made changes.